File: NovaCore-Spigot_v1_18_R1/src/main/java/net/zeeraa/novacore/spigot/version/v1_18_R1/ChunkLoaderImplementation.java
package net.zeeraa.novacore.spigot.version.v1_18_R1;

import org.bukkit.Chunk;
import net.zeeraa.novacore.spigot.abstraction.ChunkLoader;

public class ChunkLoaderImplementation extends ChunkLoader {
	public ChunkLoaderImplementation() {
		super();
	}

	@Override
	protected void onAdd(Chunk chunk) {
		chunk.setForceLoaded(true);
	}

	@Override
	protected void onRemove(Chunk chunk) {
		chunk.setForceLoaded(false);
	}
}

File: NovaCore-Spigot_v1_19_R1/src/main/kotlin/net/zeeraa/novacore/spigot/version/v1_19_R1/ChunkLoaderImplementation.kt
package net.zeeraa.novacore.spigot.version.v1_19_R1

import net.zeeraa.novacore.spigot.abstraction.ChunkLoader
import org.bukkit.Chunk

class ChunkLoaderImplementation : ChunkLoader() {
	override fun onAdd(chunk: Chunk) {
		chunk.isForceLoaded = true
	}

	override fun onRemove(chunk: Chunk) {
		chunk.isForceLoaded = false
	}
}